src/index.ts

```diff
@@ -33,10 +46,58 @@ async function serverChannel(channel: Channel, { projectToken }: { projectToken:
     });
   });
 
+  channel.on(
+    UPDATE_PROJECT,
+    async ({ projectId, projectToken: updatedProjectToken }: UpdateProjectPayload) => {
+      projectToken = updatedProjectToken;
+
+      const mainPath = await findConfig("main");
+      const MainConfig = await readConfig(mainPath);
+
+      const addonsConfig = MainConfig.getFieldValue(["addons"]);
+      const updatedAddonsConfig = addonsConfig.map(
+        (addonConfig: string | { name: string; options?: Record<string, string> }) => {
+          const fullConfig = typeof addonConfig === "string" ? { name: addonConfig } : addonConfig;
+          if (fullConfig.name === CHROMATIC_ADDON_NAME) {
+            return {
+              ...fullConfig,
+              options: { projectId, projectToken, ...fullConfig.options },
+            };
+          }
+          return addonConfig;
+        }
+      );
+
+      MainConfig.setFieldValue(["addons"], updatedAddonsConfig);
+      await writeConfig(MainConfig);
+    }
+  );
+
   return channel;
 }
 
-module.exports = {
+// TODO: use the chromatic CLI to get this info? Or should we make it core to SB?
+const execPromise = promisify(exec);
+async function getGitInfo() {
+  const branch = (await execPromise("git rev-parse --abbrev-ref HEAD")).stdout.trim();
+  const commit = (await execPromise("git log -n 1 HEAD --format='%H'")).stdout.trim();
+  return { branch, commit };
+}
+
+const config = {
   managerEntries,
   experimental_serverChannel: serverChannel,
+  // FIXME: we should only do this in dev
+  env: async (env: Record<string, string>, { projectId }: { projectId: string }) => {
+    const { branch, commit } = await getGitInfo();
```

Comment:
About the TODO:
The second argument is the `Options` object, off which you can extract the `configType` property.
That property will be either `development` or `production`.